File: RecipeBook.Web/lib/model/recipe_command.dart
import 'package:json_annotation/json_annotation.dart';

import 'ingredient.dart';

part 'recipe_command.g.dart';

@JsonSerializable(explicitToJson: true)
class RecipeCommand {
  RecipeCommand({
    this.recipeId,
    required this.title,
    required this.description,
    required this.cookingTimeInMinutes,
    required this.portionsCount,
    required this.tags,
    required this.steps,
    required this.ingredients,
  });

  final int? recipeId;
  final String title;
  final String description;
  final int cookingTimeInMinutes;
  final int portionsCount;
  final List<String> tags;
  final List<String> steps;
  final List<Ingredient> ingredients;

  factory RecipeCommand.fromJson(Map<String, dynamic> json) => _$RecipeCommandFromJson(json);

  Map<String, dynamic> toJson() => _$RecipeCommandToJson(this);
}

File: RecipeBook.Application/Services/Entities/RecipeCommand.cs
using System.Collections.Generic;
using RecipeBook.Domain.Entities;

namespace RecipeBook.Application.Services.Entities
{
    public class RecipeCommand
    {
        public RecipeCommand( int id,
            string title,
            string description,
            int cookingTimeInMinutes,
            int portionsCount,
            string userName,
            List<string> tags,
            List<string> steps,
            List<Ingredient> ingredients,
            StorageFile storageFile )
        {
            RecipeId = id;
            Title = title;
            Description = description;
            CookingTimeInMinutes = cookingTimeInMinutes;
            PortionsCount = portionsCount;
            UserName = userName;
            Tags = tags;
            Steps = steps;
            Ingredients = ingredients;
            StorageFile = storageFile;
        }

        public int RecipeId { get; }
        public string Title { get; }
        public string Description { get; }
        public int CookingTimeInMinutes { get; }
        public int PortionsCount { get; }
        public string UserName { get; }
        public List<string> Tags { get; }
        public List<string> Steps { get; }
        public List<Ingredient> Ingredients { get; }
        public StorageFile StorageFile { get; }
    }
}
